client/src/planwise/client/projects2/components/settings.cljs

```diff
@@ -410,12 +426,19 @@
         tags            (subscribe [:projects2/tags])
         regions         (subscribe [:regions/list])]
     (fn [{:keys [read-only step]}]
-      (let [project @current-project
+      (let [project         @current-project
             step-data       (first (filter #(= (:step %) step) sections))
             bbox            (:bbox (first (filter #(= (:id %) (:region-id @current-project)) @regions)))
             region-geo      (subscribe [:regions/preview-geojson (:region-id @current-project)])
             preview-map-url (if (:region-id @current-project) (static-image @region-geo map-preview-size))]
         (dispatch [:regions/load-regions-with-preview [(:region-id @current-project)]])
+        (when (nil? step-data)
+          (dispatch
+           [(if read-only :projects2/navigate-to-settings-project :projects2/navigate-to-step-project)
+            (:id project)
+            (or (when-not read-only
+                  (:step (first (filter #(not (s/valid? (:spec %) project)) sections))))
+                (:step (last sections)))]))
```

Comment:
This can be extracted to a function `infer-step` and that can make the code easier to understand.